onebusaway-android/src/main/java/org/onebusaway/android/directions/util/LocationUtil.java

```diff
@@ -0,0 +1,215 @@
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+        addresses = filterAddressesBBox(region, addresses);
+
+        boolean resultsCloseEnough = true;
+
+        if (geocodingForMarker && latLngSet) {
+            float results[] = new float[1];
+            resultsCloseEnough = false;
+
+            for (CustomAddress addressToCheck : addresses) {
+                Location.distanceBetween(latitude, longitude,
+                        addressToCheck.getLatitude(), addressToCheck.getLongitude(), results);
+                if (results[0] < GEOCODING_MAX_ERROR) {
+                    resultsCloseEnough = true;
+                    break;
+                }
+            }
+        }
+
+        if ((addresses == null) || addresses.isEmpty() || !resultsCloseEnough) {
+            if (addresses == null) {
+                addresses = new ArrayList<CustomAddress>();
+            }
+            Log.e(TAG, "Geocoder did not find enough addresses: " + addresses);
+        }
+
+        addresses = filterAddressesBBox(region, addresses);
+
+        if (geocodingForMarker && latLngSet && addresses != null && !addresses.isEmpty()) {
+            float results[] = new float[1];
+            float minDistanceToOriginalLatLon = Float.MAX_VALUE;
+            CustomAddress closestAddress = addresses.get(0);
+
+            for (CustomAddress addressToCheck : addresses) {
+                Location.distanceBetween(latitude, longitude,
+                        addressToCheck.getLatitude(), addressToCheck.getLongitude(), results);
+                if (results[0] < minDistanceToOriginalLatLon) {
+                    closestAddress = addressToCheck;
+                    minDistanceToOriginalLatLon = results[0];
+                }
+            }
+            addressesReturn.add(closestAddress);
+        } else {
+            addressesReturn.addAll(addresses);
+        }
+
+        return addressesReturn;
+    }
+
+    /**
+     * Filters the addresses obtained in geocoding process, removing the
+     * results outside server limits.
+     *
+     * @param addresses list of addresses to filter
+     * @return a new list filtered
+     */
+    private static List<CustomAddress> filterAddressesBBox(ObaRegion region, List<CustomAddress> addresses) {
+        if ((!(addresses == null || addresses.isEmpty())) && region != null) {
+            for (Iterator<CustomAddress> it = addresses.iterator(); it.hasNext(); ) {
+                CustomAddress address = it.next();
+
+                Location loc = new Location("");
+                loc.setLatitude(address.getLatitude());
+                loc.setLongitude(address.getLongitude());
+
+                if (!RegionUtils.isLocationWithinRegion(loc, region)) {
+                    it.remove();
+                }
+            }
+        }
+        return addresses;
+    }
+
+    public static LatLngBounds getRegionBounds(ObaRegion region) {
```

Comment:
Any code that references Maps API v2 objects like `LatLngBounds` must be in the `org.onebusaway.android.map.googlemapsv2` package, otherwise it will won't work on the Amazon build variant (and any other future non-Google variant).  I'd suggest trying to leave the utility methods where they are but work around the Maps v2 objects.  You can move utilities that specifically deal with Maps API v2 objects like this to `MapHelpV2`.  This might prove more difficult for some methods than others - if you hit a barrier let me know.